src/main/preference.js

```diff
@@ -15,6 +15,7 @@ class Preference {
     this.userDataPath = userDataPath
 
     if (!fs.existsSync(userDataPath)) {
+      mkdir(getPath('userData'))
```

Comment:
I suppose this will cause error too when `getPath('appData')` is also not existed?

I am newer at nodejs, I think a `ensureDir` method needed after I do some research.

```javascript
const { spawnSync } = require('child_process')

const ensureDir = dir => {
  spawnSync('mkdir', ['-p', dir])
}
// or
const fse = require('node-fs-extra')
const ensureDir = dir => {
  fse.ensureDirSync(dir)
}
```
What's your opinion?

Reply:
> I suppose this will cause error too when getPath('appData') is also not existed?

Weird... Seems to be a electron fail. Elsewise `fs.mkdirSync(...)` will create all directories.

> What's your opinion?

No this won't work on Windows and  `fs.mkdirSync(...)` uses `mkdir` on Unix.

Reply:
Ok

Reply:
> Weird... Seems to be a electron fail.

This means that if you delete your `appData` directory electron will throw an exception. Which is weird. Maybe we should check this or use a fallback directory, but actually the directory should always exist.

Reply:
I also prefer the second method, 👍 go ahead!